content/posts/2020/02/thai-in-2562/index.eng.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Software from Year 2562 Emerges in Thailand?
+date: 2020-02-25
+authors: [hong.minhee, swen.mun]
+translators: [kidon.seo]
+---
+## Guest from the Future
+
+Last December, we finally conducted our first alpha test, and thankfully, many people from all over the world participated. It was both a great opportunity and a challenge for the team. Of course, there were big and small problems. Among them, one problem that took us by surprise was IBD.
+
+<abbr title="initial block download">IBD</abbr> is a stage that occurs when you turn on a game, download blocks from other peers in the network and sync them to the latest state. Since we had participants from all over the world, IBD often took a long time or even ended abnormally due to network delays.
+
+One of our participants reported a very unusual symptom. Unlike other problems we had at that time, the participant successfully downloaded the first block but couldn’t download the rest afterwards.
+
+{{<
+figure
+  src="1.png"
+  caption="Screenshot of Our Participant"
+>}}
+
+We checked the screenshot our user sent us and realized one strange thing. The date was 2562 instead of 2019. So we came up with a hypothesis. For some reason, the file system was broken and the hash of the block header was being miscalculated.
+
+To verify this, we asked for permission to investigate the problem directly through remote desktop.
+
+
+## The Date Was Not Wrong
+
+Fortunately, the user experiencing this problem kindly accepted our request for remote access. And once we were able to access the desktop, we saw that the date was still in 2562 as it appeared in the screenshot. First, we opened the Control Panel to sync time. Like most modern operating systems, Windows has the ability to synchronize time over the network. But when we resynchronized time, the control panel and the system's year didn't change and remained in 2562. Notably, apart from the year, date and time were no different from the system on our side.
+
+We decided to explore the problem a little bit more, and while checking out the control panel, one detail caught our attention.
+
+{{<
+figure
+  src="2.png"
+  caption="Year Displayed as 2019 in Gregorian calendar"
+>}}
+
+No one in our development team could read Thai, but looking at "Date in Gregorian”, we noticed that "2562" was a different expression of "2019." So we changed the format, and as expected, December 16th, 2019 was displayed. And when we launched the game for testing, it went smoothly at the IBD stage.
```

Comment:
```suggestion
No one in our development team could read Thai, but looking at "Date in Gregorian”, we noticed that "2562" was a different expression of "2019". So we changed the format, and as expected, December 16th, 2019 was displayed. And when we launched the game for testing, it went smoothly at the IBD stage.
```

Reply:
@moreal It's the matter of style; see also <https://en.wikipedia.org/wiki/Quotation_marks_in_English#Order_of_punctuation>.  As we basically follow American orthography, to me the previous use of quotes is more consistent…

Reply:
Oh, I didn't know... Next time, I will remember and review these parts.